### DISCUSSION TITLE
How to detect features actually in use

### DISCUSSION THREAD

**Discussion by goatfryed** [TanStack/table#5342] [Q&A]
Is there a way to check which features are actually in use at the moment? With in use, i mean that a RowModel was provided.

In my project, I have various tables that all had a similar layout. So far, all of them used the same feature set, but I wanted to remove pagination and/or filtering from some of them.
Now, in my view component I'd like to check whether the given table instance has pagination configured and only display pagination or filters, if it's in use, but I couldn't find a way.
I'm aware that the API allows me to call the related methods and actually manages the state, but of course the rows won't change.

I settled to add a meta attribute to the table for now that I use to omit pagination or filters, but it feels redundant to the model declarations.

Is there a nicer way that I missed?

**Comment by goatfryed** [comment]
I just found out that this duplicated #5306. Since my question is a bit more general but the other one is a bit more detailed, I'm cross referencing and will ask the other author to merge this into one question

**Reply by marceloverdijk** [reply]
It seems something trivial, but nevertheless not clear how to check what is enabled or not.
I'm also building a generic React table component for one of my projects which I want to pass Tanstack table instance and that based on which features are enables render certain functionality.
